### ISSUE TITLE
Plug-in export error with lombok

### ISSUE THREAD

**Issue by michaelansley** [archi-contribs/database-plugin#115]
Hi Hervé

I've imported the plug-in source to my eclipse dev environment, and got it all working along with the rest of Archi.  If I run Archi in eclipse, I will be able to run the import/export functions of the database plug-in, I've made some extremely minor test changes and they work when running from the IDE.  I want to export a jar file so that I can install it into a test environment, and so I used the File | Export... function and generated the output, at the end of which I got a message saying that there were errors and providing a log file.  At the same time, a jar file was produced.
Following is the (partial) contents of the error log file:

----------
1. ERROR in /home/xxxxxxx/git/database-plugin/v2/sources/src/org/archicontribs/database/preferences/DBTextFieldEditor.java (at line 0)
	/**
	^
Internal compiler error: java.util.ServiceConfigurationError: javax.annotation.processing.Processor: Provider lombok.launch.AnnotationProcessorHider$AnnotationProcessor could not be instantiated at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582)
----------
1 problem (1 error)

I've done some Googling, but so far no success in finding the cause or possible solutions.

Any ideas what is causing this, or how I might resolve it?

I should add that the jar file produced does not work as a plug-in to an installation of Archi.  Although it appears in the plug-ins list (Help | Manage Plug-ins...), the preferences page fails with an error and there are no File | Import... or File | Export... entries.

Regards,
Michael

**Comment by michaelansley** [comment]
I should also say that I installed lombok by 'running the jar' and using the installer.  After lombok was installed, development (editor) and execution worked fine.  Just the export is problematic.

**Comment by herve91** [comment]
Hi Michael,

The Lombok project is effectively needed to ease the declaration of some classes or methods, as every dev, I'm quite lazy :)

What error do you get now that you fixed Lombok ?

Best regards
Hervé

**Comment by herve91** [comment]
BTW, to export the plugin as a JAR file, you need to right click on the project in Eclipse, then select "Export ..." and then "Plugin-in development / Deployable plug-ins and fragments". than, just select a directory where Eclipse will save the JAR file.

Best regards
Hervé

**Comment by michaelansley** [comment]
The error I posted above happened after installing lombok (using the jar installer).

To try and get the plug-in jar, I followed exactly the instructions you put above.

For now, still the same problem, and I've continued searching for solutions, but no joy.  I also deleted the database plug-in project from eclipse, rebuilt, then added the plug-in project again (the idea being that the first time I added the project, lombok wasn't installed) and then rebuilt, but still no solution.

**Comment by herve91** [comment]
Difficult to help you with this. I do not have issues with Lombok on my side.

What I can suggest is to check if the installation did well:

* check that the Lombok.jar file has been correctly copied to the Eclipse folder, the same folder where the eclipse.exe file stands (C:\Users\Herve\eclipse\rcp-2020-06\eclipse on my PC)
* check that the Lombok jar file is correctly defined in the eclipse.ini file (there muse be a line like "-javaagent:C:\Users\Herve\eclipse\rcp-2020-06\eclipse\lombok.jar", adapted to your local folder of course)

**Comment by michaelansley** [comment]
OK, thanks Hervé.  I will try it on Windows, as so far I have only tried it in a Linux VM that I use for specific kinds of dev work.

I will also try using a previous version of lombok... what version are you using?

**Comment by michaelansley** [comment]
I tried downgrading the lombok implementation to 1.18.16 (I was previously using 1.18.18) and now I can export the plug-in jar, no problem.  I tested it in a clean Archi installation on Linux and also on Windows (both Archi 4.8.1) and it worked.

I also updated the plug-in to ensure that dialogs open on the same monitor as Archi (scratching my own personal itch, as it were).

So, it appears that there is some kind of issue in lombok 1.18.18 used with eclipse 4.18 and ubuntu linux and DBPlug-in 2.2.16 source.
